packages/sitecore-jss-angular/src/lib.module.ts

```diff
@@ -81,14 +82,24 @@ export class JssModule {
   }
 
   /** Instantiates a module for a lazy-loaded JSS component */
-  static forChild(component: Type<any>): ModuleWithProviders {
+  static forChild(components: Type<any> | Array<Type<any>>): ModuleWithProviders {
+    const providers: Provider[] = [
+      { provide: ROUTES, useValue: [], multi: true },
+    ];
+
+    if (Array.isArray(components)) {
+      components.forEach((component) => {
+        providers.push({ provide: ANALYZE_FOR_ENTRY_COMPONENTS, useValue: component, multi: true });
+        providers.push({ provide: DYNAMIC_COMPONENT, useValue: component });
```

Comment:
That makes sense. I vaguely recall attempting to prototype placing several lazy components in a module and there was some reason why it was less than simple.

Reply:
Yes, you are right. Sorry for my mistake. I made other change and start to use ComponentNameAndType in forChild because I need to have a name of the component in jss-component-factory.service. With help of component name we success loaded the correct type like is made for loaded component.